src/vm/memory/segments.zig

```diff
@@ -797,6 +826,46 @@ test "MemorySegmentManager: getSegmentSize should return null if missing segment
     try expectEqual(@as(?u32, null), memory_segment_manager.getSegmentSize(3));
 }
 
+test "MemorySegmentManager: relocateSegments for one segment" {
+    var memory_segment_manager = try MemorySegmentManager.init(std.testing.allocator);
+    defer memory_segment_manager.deinit();
+    try memory_segment_manager.segment_used_sizes.put(0, 1);
+    const actual_value = try memory_segment_manager.relocateSegments(std.heap.page_allocator);
```

Comment:
In tests, we should always use `std.testing.allocator`

Reply:
it doesn't work for arrays

Reply:
I read somewhere. I think on zig reddit that it's better to return a slice when you no longer want to manipulate a value. And, slices are more straightforward to use that `ArrayList`.